ark-activemq/values.yaml

```diff
@@ -0,0 +1,99 @@
+# Default values for ark-activemq.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+# Please leave the `replicaCount` to 1 for the time being. Clustering is not
+# supported yet.
+replicaCount: 1
+
+image:
+  repository: 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_activemq
+  pullPolicy: IfNotPresent
+  tag: latest
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+# Console username and password: Please set a secure password
+adminUsername: admin
+adminPassword: admin
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: ""
+
+podAnnotations: {}
+
+podSecurityContext: {}
+  # fsGroup: 2000
+
+securityContext: {}
+  # capabilities:
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+# Uncomment ports that you want ActiveMQ to open. You can also pass the ports
+# you want to open as arguments to the `helm` program. Please note you need to
+# open at least one of the following ports: amqp, mqtt, openwire, websocket or
+# stomp.
+ports:
+  metrics: 5556
+#  console: 8161
+#  amqp: 5672
+#  mqtt: 1883
+#  openwire: 61616
+#  websocket: 61614
+#  stomp: 61613
```

Comment:
@fabricetriboix why are ports are commented. Here is what  found in arkcase-ce activemq config
```
  <transportConnectors>
            <!-- DOS protection, limit concurrent connections to 1000 and frame size to 100MB -->
            <transportConnector name="openwire" uri="ssl://0.0.0.0:61616?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
            <transportConnector name="amqp" uri="amqp://0.0.0.0:5672?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
            <transportConnector name="stomp" uri="stomp+ssl://0.0.0.0:61613?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
            <transportConnector name="mqtt" uri="mqtt://0.0.0.0:1883?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
            <transportConnector name="ws" uri="ws://0.0.0.0:61614?maximumConnections=1000&amp;wireFormat.maxFrameSize=104857600"/>
        </transportConnectors>
```

Reply:
@naveen-armedia Ports are commented by default. The script that will install the ActiveMQ Helm chart will provide its own values and thus open only the necessary ports.

Reply:
@fabricetriboix well all the keys defined in the Values.yaml could be overridden at runtime either manually  or through the script having a separate Values.yaml. This is the flexibility we have with Helm. But having those defined here would be more readable for any starters..

Reply:
@naveen-armedia The principle of least privilege is to not enable something unless it is required. I think the Helm chart should not make assumptions about which ports are going to be used or not. I wrote a clear comment above explaining how ports can be open. I believe a port should remain close unless and until it is required to be open.

Reply:
@fabricetriboix  please check my comments above. keys in Values.yaml file is always be overridden from parent chart or using separate Values.yaml at runtime. Also its does not mean if we uncomment keys we should use them at runtime.

Reply:
@naveen-armedia You are mistaken. Helm merges all the sources for its values, be it the default `values.yaml` file in the Helm chart, any values files passed on the `helm install` command line and any values passed directly on the `helm install` command line. So if I leave, say, the `console` port uncommented in the chart's default `values.yaml` file, but comment it or delete it from a custom values file passed on the command line, the resulting values that Helm will use will still have the `console` port specified. I just tried it to be 100% sure. You can try it for yourself if you want.